cci_cameroon/analysis/text_similarity/similarity_search_sample.py

```diff
@@ -0,0 +1,608 @@
+
+# %%
+print(
+    d, p
+)  # the first sentence has a great match and the eucledian distance is smaller - 27.04
+
+# %%
+# another test case
+samp2 = ["Avoir un téléphone peux attirer les soucis!"]
+samp2_emb = model_fr.encode(samp2, convert_to_tensor=True)
+
+# %%
+cosine_scoresb = util.pytorch_cos_sim(samp2_emb, embeddings11)
+print(cosine_scoresb)
+
+# %%
+# Using the FAISS flat index to compute the distance.
+samp2_emb = model_fr.encode(samp2)
+d, p = fr_index1.search(samp2_emb, 2)
+print(d, p)
+
+# %%
+
+# %% [markdown]
+# # The LaBSE pre-trained model
+# It is a multilingual model pre-trained in 109 language. We use it for similarity match search of french text
+
+# %%
+from transformers import BertModel, BertTokenizerFast
+
+# %%
+tokenizer = BertTokenizerFast.from_pretrained("setu4993/LaBSE")
+model = BertModel.from_pretrained("setu4993/LaBSE")
+model = model.eval()
+
+# %%
+fr_sentences = [
+    "la chaleur tue le coronavirus",
+    "le virus se trans,et d'un contact à un autre part les mains",
+    " le soleil tue le virus",
+    "LE Nord ne peut pas avoir le virus. il fait trop chaud",
+    "Le virus se transmet à travers les gouttelettes qui sortent du nez et de la bouche",
+    "Aucun groupe dans notre communauté n'est responsable de la propagation du virus",
+]
+fr_search = ["Le virus ne résiste pas à la chaleur"]
+fr_inputs = tokenizer(fr_sentences, return_tensors="pt", padding=True)
+fr_search_input = tokenizer(fr_search, return_tensors="pt", padding=True)
+with torch.no_grad():
+    fr_outputs = model(**fr_inputs)
+    fr_search_output = model(**fr_search_input)
+# extract the embeddings
+fr_embeddings_out = fr_outputs.pooler_output
+fr_embedding_search = fr_search_output.pooler_output
+
+# %%
+# To calculate similarity between them, we compute the L2 norms of the embeddings
+import torch.nn.functional as F
+
+
+def similarity(embeddings_1, embeddings_2):
+    normalized_embeddings_1 = F.normalize(embeddings_1, p=2)
+    normalized_embeddings_2 = F.normalize(embeddings_2, p=2)
+    return torch.matmul(
+        normalized_embeddings_1, normalized_embeddings_2.transpose(0, 1)
+    )
+
+
+# %%
+print(similarity(fr_embedding_search, fr_embeddings_out))
+
+# %%
+cosine_scoresb2 = util.pytorch_cos_sim(fr_embedding_search, fr_embeddings_out)
+print(cosine_scoresb2)
+
+# %%
+# compute cosine similarity using the sentences used for french_semantic model
+# names of variables are : sentences1 , sample1
+sentences1_fr_inputs = tokenizer(sentences_french, return_tensors="pt", padding=True)
+sample1_fr_input = tokenizer(sample_french1, return_tensors="pt", padding=True)
+sample2_fr_input = tokenizer(sample_french2, return_tensors="pt", padding=True)
+with torch.no_grad():
+    sentences1_fr_outputs = model(**sentences1_fr_inputs)
+    sample1_fr_output = model(**sample1_fr_input)
+    sample2_fr_output = model(**sample2_fr_input)
+# extract the embeddings
+sentences1_embeddings_out = sentences1_outputs.pooler_output
```

Comment:
```
---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
<ipython-input-88-f4adebf9de69> in <module>()
      9     sample2_fr_output = model(**sample2_fr_input)
     10 # extract the embeddings
---> 11 sentences1_embeddings_out = sentences1_outputs.pooler_output
     12 sample1_embedding_search = sample1_output.pooler_output

NameError: name 'sentences1_outputs' is not defined
```

Reply:
I got the above error trying to run this line

Reply:
Looks like the variable needs to be `sample1_fr_output`

Reply:
Unable to reproduce it at my end Izzy.